### DISCUSSION TITLE
How to use an existing postgres database

### DISCUSSION THREAD

**Discussion by Tarang** [geldata/gel#2674] [Q&A]
How can I set up edgedb so it uses a postgres database that already exists? Is there any documentation that explains how this works?

As far as I can tell the current tutorials and configuration handles & manages a postgres instance behind the scenes for you.

**Comment by elprans** [comment]
One can use the `--postgres-dsn=<libpq-compatible-uri>` command-line argument when starting `edgedb-server` or specify it via the `EDGEDB_POSTGRES_DSN` if using the [Docker image](https://github.com/edgedb/edgedb-docker).  We'll be adding proper documentation for split-Postgres production deploys soon.

**Reply by clmnin** [reply]
I think Tarang was taking about using an existing deployment - using EdgeDB as an ORM of sorts.

Maybe you should clarify *existing Postgres* databases can't flip a switch and start using EdgeDB. That postgres is used as a data store.

**Reply by Tarang** [reply]
Thats right. I was wondering if we can use Edgedb and Postgres. If we use Edgedb does it mean we cant use the postgres part directly anymore?

**Reply by elprans** [reply]
EdgeDB can run on an existing PostgreSQL cluster (via `--postgres-dsn`), but you are not supposed to modify the actual PostgreSQL databases managed by EdgeDB directly.  You can continue to use the non-EdgeDB databases in the cluster as regular Postgres databases, of course.

**Reply by korjavin** [reply]
I try using `podman run --name=edgedb --pod edge -e EDGEDB_POSTGRES_DSN="username=postgres host=pg db=edgedb" -e EDGEDB_SERVER_SECURITY=insecure_dev_mode edgedb`

but it creates another instances of pg instead of using one I provided

**Reply by elprans** [reply]
Since the comment above was posted, the option was renamed to `EDGEDB_SERVER_POSTGRES_DSN`.  Also, it should be in the URI format: `postgres://user@host/db`

**Reply by korjavin** [reply]
Thank you! It works this way

**Comment by RashiqAzhan** [comment]
So, currently, it is impossible to switch to edgedb with an existing postgres instance with production data?

Is there any way to have existing infrastructure to use the current pg instance while newer services use the edgedb layer to access and modify the same pg instance?
